packages/office-ui-fabric-react/src/components/Fabric/Fabric.base.tsx

```diff
@@ -35,17 +35,23 @@ export class FabricBase extends React.Component<
 
   public render() {
     const { as: Root = 'div', theme, dir } = this.props;
-    const classNames = this._getClassNames();
-    const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties);
     const isRTL = dir === 'rtl';
-    let renderedContent = <Root {...divProps} className={classNames.root} ref={this._rootElement} />;
 
-    // Expose an rtl based theme if dir is specified and it doesn't agree with the theme setting.
+    // Adjust dir if prop is specified and it doesn't agree with the theme setting.
     if (dir !== undefined && theme && (theme.rtl === undefined || theme.rtl !== isRTL)) {
-      renderedContent = <Customizer settings={{ theme: getRTLTheme(theme, isRTL) }}>{renderedContent}</Customizer>;
+      const newTheme = getRTLTheme(theme, isRTL);
+      const classNames = this._getClassNames(newTheme);
+      const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties);
+      return (
+        <Customizer settings={{ theme: newTheme }}>
+          <Root {...divProps} className={classNames.root} ref={this._rootElement} />
+        </Customizer>
+      );
+    } else {
+      const classNames = this._getClassNames(this.props.theme!);
+      const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties);
+      return <Root {...divProps} className={classNames.root} ref={this._rootElement} />;
```

Comment:
Isn't `divProps` the same in both branches? It only seems to depend on `this.props`, and I'm not seeing how that would differ in each conditional branch.

If you do have to repeat yourself, could you do:
```tsx
      const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties);
      const rootContent = <Root {...divProps} className={classNames.root} ref={this._rootElement} />;
      return (
        <Customizer settings={{ theme: newTheme }}>
          {rootContent}
        </Customizer>
      );
    } else {
      const classNames = this._getClassNames(this.props.theme!);
      return rootContent;
    }
```

Reply:
Err, I see now `classNames` differs for each, but it could also be a function that returns the same content applying `classNames`. (I also missed an `if` statement there.)

Reply:
I edited the original code comment to show what I meant. :)

Reply:
Yes, this would work, but I think the creation of a lambda just to take in `rootClassName` makes the intent less clear, rather than more. As written, everything that's happening is pretty straightforward.

Reply:
Well, I think at least `divProps` could be centralized. I'm not sure why that's duplicated and actually obfuscated intent when I reviewed it as I could not see why there were two instances of it.